org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/tool/EclipseFileManager.java

```diff
@@ -1421,15 +1430,19 @@ public void setLocationFromPaths(Location location, Collection<? extends Path> p
 						DefaultErrorHandlingPolicies.proceedWithAllProblems(),
 						compilerOptions,
 						new DefaultProblemFactory());
-			for (Path path : paths) {
-				List<Classpath> mp = ModuleFinder.findModules(path.toFile(), null,
-						new Parser(problemReporter, true), null, true, this.releaseVersion);
-				for (Classpath cp : mp) {
-					Collection<String> moduleNames = cp.getModuleNames(null);
-					for (String string : moduleNames) {
-						Path p = Paths.get(cp.getPath());
-						setLocationForModule(location, string,  Collections.singletonList(p));
+			for (P path : paths) {
+				try {
+					List<Classpath> mp = ModuleFinder.findModules(toFile.apply(path), null,
+							new Parser(problemReporter, true), null, true, this.releaseVersion);
+					for (Classpath cp : mp) {
+						Collection<String> moduleNames = cp.getModuleNames(null);
+						for (String string : moduleNames) {
+							Path p = Paths.get(cp.getPath());
+							setLocationForModule(location, string,  Collections.singletonList(p));
+						}
 					}
+				} catch (IllegalArgumentException iae) { // e.g., from ModuleFinder.scanForModule(Classpath, File, Parser, boolean, String)
+					// FIXME ignore for now
```

Comment:
CompilerToolJava9Tests.testBug566749() expects a regular compile error, which now additionally surfaces as the IllegalArgumentException caught here.